### DISCUSSION TITLE
RISC V support

### DISCUSSION THREAD

**Discussion by ddscentral** [SoftEtherVPN/SoftEtherVPN#2063] [General]
Just cross-compiled SoftEther Stable for a RISC V board (riscv64) and it works fine. Don't see RISC V support mentioned in the documentation. Is this just an oversight or it has not been thoroughly tested ?

I see hamcore packager still isn't updated for cross-compiling though, hamcore.se2 needs to be prebuilt on the host machine before compiling and Makefile tweaked to use the prebuilt file.

**Comment by chipitsine** [comment]
well, there was some usage in OpenWRT, I guess mostly RISC

**Reply by chipitsine** [reply]
if you are interested, I think we can introduce some RISC based CI/CD using https://github.com/vmactions (or similar)

**Reply by ddscentral** [reply]
Not sure what you meant by "mostly RISC". OpenWRT is mostly ARM nowadays. There seems to be a RISC V port though.

**Reply by chipitsine** [reply]
I thought it is RISC/MIPS. but I may be wrong on that

**Reply by ddscentral** [reply]
RISC is more of a concept than an architecture. ARM, MIPS and RISC V are all RISC architectures.
Most OpenWRT devices nowadays have switched to ARM, some 32-bit, some 64-bit. Previously, they pretty much all used 32-bit MIPS.

**Comment by chipitsine** [comment]
if you have an appetite, you can try https://cloud-v.co/ (if not already)

**Comment by ddscentral** [comment]
Thanks for the link.
So it's about not having the CI/CD setup for building RISC V packages ?

Well, I only have one RISC V board right now and it uses buildroot, not even a real distribution.
Likely would need a QEMU emulated machine for testing with a proper distro installed. That's not a problem to setup. 
The main issue is free time, that is lack of it.

I'll see what I can do.
